libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -979,3 +1019,129 @@ eth::AssemblyPointer ContractCompiler::cloneRuntime() const
 	a << u256(0x20) << u256(0) << Instruction::RETURN;
 	return make_shared<eth::Assembly>(a);
 }
+
+void ContractCompiler::addScopedVariable(VariableDeclaration const& _decl)
+{
+	// TODO check > 17
+	m_scopedVariables[_decl.scope()].push_back(&_decl);
+	if (!m_loops.empty())
+		m_loopScopedVariables[m_loops.back()].insert(&_decl);
+	appendStackVariableInitialisation(_decl);
+}
+
+void ContractCompiler::popBlockScopedVariables(ASTNode const* _node)
+{
+	for (auto _decl : m_scopedVariables[_node])
+	{
+		if (!m_loops.empty())
+			m_loopScopedVariables[m_loops.back()].erase(_decl);
+		CompilerContext::LocationSetter locationSetter(m_context, *_decl);
+		CompilerUtils(m_context).popStackElement(*_decl->type());
+		m_context.removeVariable(*_decl);
+	}
+	m_scopedVariables.erase(_node);
+}
+
+/**
+Whenever a break or continue is reached, it counts the number of
+stack slots that should be freed at that moment with respect to
+local variables, and creates a new tag that leads to the freeing
+of those variables.
+All the break/continue statements inside the same loop share this
```

Comment:
While this is a clever technique to save some code, this is disastrous for bytecode analysis in the optimizer and other tools. Also I don't think it happens too often that we have multiple break / continue statements in a loop where this would be beneficial.

Reply:
Ok ;)
Confirming then: each break/continue/return should point to their own non-shared structure that jumps accordingly at the end.